kolibri/core/assets/src/heartbeat.js

```diff
@@ -39,17 +40,26 @@ class HeartBeat {
   setInactive() {
     this.active = false;
   }
+  wait() {
+    this.timerId = setTimeout(this.beat, this.delay);
+    return this.timerId;
+  }
   beat() {
     if (this.active) {
       logging.debug('There was activity, polling session endpoint!');
       this.kolibri.resources.SessionResource.getModel('current').fetch({}, true).catch((error) => {
         logging.error('Periodic server polling failed, with error: ', error);
       });
       this.setActivityListeners();
+    } else {
+      logging.debug('No user activity, polling keepalive endpoint to keep session active');
+      this.kolibri.resources.KeepAliveResource.getModel('now').fetch({}, true).catch((error) => {
```

Comment:
Personally, I would have just done a straight ajax call here rather than leveraging the API Resources system. (the `keepalive` endpoint isn't a resource with state to retrieve or set.)

Reply:
Yes, the only reason I did this was for the convenience of the pre-existing system.